src/main/java/com/robothand/highqualitybot/permission/PermissionGroup.java

```diff
@@ -0,0 +1,98 @@
+package com.robothand.highqualitybot.permission;
+
+import com.robothand.highqualitybot.Bot;
+import com.robothand.highqualitybot.Utils;
+import com.robothand.highqualitybot.command.Command;
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Role;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Hashtable;
+
+/**
+ * Represents a set of permissions for a group of Users
+ */
+public class PermissionGroup {
+    private ArrayList<Command> allowed;
+    private ArrayList<Command> disallowed;
+    private boolean allowedHasPrecedence;
+    private int priority;
+    private Role role;
+
+    public PermissionGroup(String filename) throws FileNotFoundException {
+        // default values
+        allowed = new ArrayList<>();
+        disallowed = new ArrayList<>();
+        allowedHasPrecedence = true;
+        priority = 0;
+
+        Hashtable<String,String> table = Utils.readCFG(filename);
+
+        for (String key : table.keySet()) {
+            String value = table.get(key);
+
+            switch (key) {
+                case "allowed":
+                    allowed = PermissionManager.instance().parseCommandList(value);
+                    break;
+
+                case "disallowed":
+                    disallowed = PermissionManager.instance().parseCommandList(value);
+
+                case "allowedHasPrecedence":
```

Comment:
I think you could replace this whole statement with `allowedHasPrecedence = value.equals("true");`